horovod/tensorflow/__init__.py

```diff
@@ -202,7 +219,18 @@ def after_create_session(self, session, coord):
 
 @_cache
 def _make_allreduce_grads_fn(name, device_dense, device_sparse,
-                             compression, sparse_as_dense, op):
+                             compression, sparse_as_dense, op, gradient_predivide_factor):
+    if op == Average and gradient_predivide_factor != 1.0:
```

Comment:
If it's alright with you, my preference would be to leave the MXNet code as is (averaging in the C++ code, now for ROCm only) since that is how it was originally implemented and will be removed eventually anyway.